assets/js/translator.js

```diff
@@ -4,10 +4,39 @@ var curSrcLang, curDstLang;
 var recentSrcLangs = [], recentDstLangs = [];
 var droppedFile;
 var textTranslateRequest;
+var recaptchaScriptSrc = "https://www.google.com/recaptcha/api.js?onload=recaptchaRenderCallback&render=explicit&hl=";
+
+var recaptchaRenderCallback = function () {
+    grecaptcha.render('suggestRecaptcha', {
+      'sitekey' : config.SUGGESTIONS.recaptcha_site_key
+    });
+}
 
 if(modeEnabled('translation')) {
     $(document).ready(function () {
         synchronizeTextareaHeights();
+        var localLang = iso639Codes[$('.localeSelect').val()];
+        var newSrc = recaptchaScriptSrc + 'en';
+        var backoff = false;
+
+        for(var i = 0; i < localizeRecaptcha.length; i++) {
+            if (localLang == localizeRecaptcha[i]) {
+                newSrc = recaptchaScriptSrc + localLang;
+                backoff = false;
+                break
+            }
+            else {
+                backoff = true;
+            } 
+        }
+
+        if (backoff) {
+            newSrc = recaptchaScriptSrc + localizeRecaptchaAlternative[localLang]; 
+        }
+
+        $('#recapscript').attr('src', newSrc);
+        $('#recapscript').attr('async', '');
```

Comment:
That's why they're recommended when the script is loaded normally via HTML.
When you add it dynamically, I'm not sure these attributes achieve
anything. Can you check if it works without them?

On Aug 25, 2016 7:18 AM, "Kira Droganova" <notifications@github.com> wrote:

> In assets/js/translator.js
> <https://github.com/goavki/apertium-html-tools/pull/68#discussion_r76231772>
> :
>
> > +            if (localLang == localizeRecaptcha[i]) {
> > +                newSrc = recaptchaScriptSrc + localLang;
> > +                backoff = false;
> > +                break
> > +            }
> > +            else {
> > +                backoff = true;
> > +            }
> > +        }
> > +
> > +        if (backoff) {
> > +            newSrc = recaptchaScriptSrc + localizeRecaptchaAlternative[localLang];
> > +        }
> > +
> > +        $('#recapscript').attr('src', newSrc);
> > +        $('#recapscript').attr('async', '');
>
> Add attributes async and defer
> Mans at https://developers.google.com/recaptcha/ says that I need them
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/goavki/apertium-html-tools/pull/68/files/f4ebebee15e2453007e977dc5a3753dc0ea65f92#r76231772>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEBEft6BmXwbumg-o3SIG_pfkZJrU2_cks5qjYgLgaJpZM4JpRx5>
> .
>